src/chips/nrf51822/Makefile.mk

```diff
@@ -0,0 +1,19 @@
+ARCH = cortex-m0
+
+RUSTC_FLAGS += -C opt-level=3 -Z no-landing-pads
+RUSTC_FLAGS += --target $(SRC_DIR)chips/$(CHIP)/target.json
+RUSTC_FLAGS += -Ctarget-cpu=$(ARCH) -C relocation_model=static
+RUSTC_FLAGS += -g -C no-stack-check
+
+CFLAGS += -g -O3 -std=gnu99 -mcpu=$(ARCH) -mthumb -nostdlib -T$(SRC_DIR)chips/$(CHIP)/loader.ld
+LDFLAGS += -mcpu=$(ARCH) -mthumb
+LOADER = $(SRC_DIR)chips/$(CHIP)/loader.ld
```

Comment:
I'd suggest TOCK_CHIP and TOCK_ARCH; that way you can pull them from environment variables if needed.